duffy/legacy/main.py

```diff
@@ -29,106 +31,160 @@
 )
 
 
-@app.post("/Node/get")
-async def get_node(data: NodeGetModel):
-    username, password, dest = obtain_credentials(data.key)
-    response = httpx.post(
-        "%s/api/v1/sessions" % dest,
-        json={
-            "nodes_specs": [
-                {
-                    "type": "physical",
-                    "quantity": data.count,
-                    "distro_type": "centos",
-                    "distro_version": data.ver,
-                },
-            ],
-        },
-        auth=(username, password),
-    )
-    respdict = response.json()
-    if response.status_code == HTTP_201_CREATED:
-        legacy_result = {
-            "ssid": respdict["id"],
-            "hosts": [respdict["nodes"][indx]["hostname"] for indx in respdict["nodes"]],
-        }
-        return legacy_result
-    elif response.status_code == HTTP_422_UNPROCESSABLE_ENTITY:
-        return JSONResponse(status_code=200, content="Failed to allocate nodes")
-    elif (
-        response.status_code == HTTP_401_UNAUTHORIZED or response.status_code == HTTP_403_FORBIDDEN
-    ):
-        return JSONResponse(status_code=403, content={"msg": "Invalid duffy key"})
+@app.get("/Node/get")
+async def get_node(
+    ver: int = None,
+    arch: str = None,
+    count: int = None,
+    key: str = None,
+    flavor: str = None,
+    cred: Credentials = Depends(obtain_credentials),
+):
+    if ver and arch and count and key and flavor:
+        response = httpx.post(
+            "%s/api/v1/sessions" % obtain_destination(),
+            json={
+                "nodes_specs": [
+                    {
+                        "type": "physical",
+                        "quantity": count,
+                        "distro_type": "centos",
+                        "distro_version": ver,
+                    },
+                ],
+            },
+            auth=(cred.username, cred.password),
+        )
+        respdict = response.json()
+        if response.status_code == HTTP_201_CREATED:
+            legacy_result = {
+                "ssid": respdict["id"],
+                "hosts": [respdict["nodes"][indx]["hostname"] for indx in respdict["nodes"]],
+            }
+            return legacy_result
+        elif response.status_code == HTTP_422_UNPROCESSABLE_ENTITY:
+            return JSONResponse(status_code=200, content="Failed to allocate nodes")
+        elif (
+            response.status_code == HTTP_401_UNAUTHORIZED
+            or response.status_code == HTTP_403_FORBIDDEN
+        ):
+            return JSONResponse(status_code=403, content={"msg": "Invalid duffy key"})
+        else:
+            return JSONResponse(status_code=200, content="Failed to allocate nodes")
     else:
-        return JSONResponse(status_code=200, content="Failed to allocate nodes")
+        return JSONResponse(status_code=200, content="Some parameters are absent")
```

Comment:
But `status_code == 200` means "everything is ok", is this a legacy Duffy bug which we're replicating here? Also, let's use symbolic status codes (from `starlette.status`) consistently.